app/services/tag_archiver.rb

```diff
@@ -17,7 +17,7 @@ def archive
       update_tag
       setup_redirects
       republish_tag
-      unsubscribe_from_email_alerts if tag.is_a?(Topic)
+      unsubscribe_from_email_alerts if tag.is_a?(Topic) && tag.level_two?
```

Comment:
I've now simplified this `if` statement with methods in the respective models.

For some reason I'm not sure about moving the `if` statement to a `return` block in the #unsubscribe_from_email_alerts definition. 
It feels like the service would call #unsubscribe_from_email_alerts unnecessarily and it would slightly breach the single responsibility principle (with the added check). We also have other examples of this pattern in services in this app. I'd be interested to know why you think it's better and reconsider changing to that pattern

Reply:
Makes sense. Let's keep that logic together 👍 
Replaced the `if tag.can_have_email_subscriptions?` with `return unless tag.can_have_email_subscriptions?` in the #unsubscribe_from_email_alerts.